example/basic/basic_hcube.cxx

```diff
@@ -23,36 +23,88 @@
 #include <sc_options.h>
 #include <sc_refcount.h>
 #include <t8_schemes/t8_default_cxx.hxx>
+#include <t8_schemes/t8_default/t8_dtri.h>
+#include <t8_schemes/t8_default/t8_dtet.h>
+#include <t8_schemes/t8_default/t8_dprism.h>
+#include <t8_schemes/t8_default/t8_dpyramid.h>
 #include <t8_forest.h>
 #include <t8_cmesh_vtk.h>
 
+static int
+t8_basic_hypercube_refine(t8_forest_t forest, t8_forest_t forest_from,
+                          t8_locidx_t which_tree, t8_locidx_t lelement_id,
+                          t8_eclass_scheme_c * ts, int num_elements,
+                          t8_element_t * elements[])
+{
+    int         type, level;
+    level = ts->t8_element_level(elements[0]);
+    if (level >= *(int *) t8_forest_get_user_data (forest) || level == 0) {
+      return 0;
+    }
+    switch(ts->t8_element_shape(elements[0])){
+    case T8_ECLASS_TRIANGLE:
+        type = ((t8_dtri_t *)elements[0])->type;
```

Comment:
We can now use the new general element function here and hence do not need to expose our implementation (which we should not do, especially not in a basic example).